### DISCUSSION TITLE
RS485 hat + RasPi 3B outdoor HW

### DISCUSSION THREAD

**Discussion by gmorganVA** [tagyoureit/nodejs-poolController#590] [General]
I am looking to add a raspberrypi 3B v1.2 to my pool automation.  Saw the recommendations for USB to RS485 but wondered if a RS485 hat would be more reliable?  Was considering putting the pi in a IP67 rated enclosure and figure I need to run the RS485 line and some sort of power line from my existing control box / circuit panel to this enclosure. (Plan to use wifi to link to my home network...my AP has really good range)

Any recommendations on the power supply / RS485 hat?  With all the electrical noise in pool equipment want to be really careful how I wire this all up.  Don't want to fry anything.

**Comment by tagyoureit** [comment]
The hat should be fine, but we don't have reliability problems with the USB adapters.  Most of the RS485 hats don't stack nicely (aka not the same form factor) as the Pi.  If you are going to be doing any more automation than just using the RS485 then you might consider the MEGA-BAS hat which has RS485 built-in (see wiki link).  We have recommendations on the [Nixie pool controller wiki](https://github.com/tagyoureit/nodejs-poolController/wiki/DIY-Standalone-Nixie-Pool-Controller#budget) and the preferred "budget" power supply is the Meanwell MDR-40-5 (most ppl use DIN rail setups because they are very flexible).

**Reply by gmorganVA** [reply]
Decided to pick up one of the USB adapters.  Will see what I can accomplish with it.  Thanks for the advice and the quick response.
